HdmiCecSource/HdmiCecSource.cpp

```diff
@@ -787,8 +787,16 @@ namespace WPEFramework
                         powerState = 0; 
                     }
                     else
+		    {
                         powerState = 1;
-
+		    }
+	            if(param->data.state.newState == IARM_BUS_PWRMGR_POWERSTATE_STANDBY)
```

Comment:
what if device new state as DEEPSLEEP? can this logic fit in to above if else at line 785? also where do you send activesource true when device transition to ON state?

Reply:
@apatel859, as per my understanding, we just need to send out the STANDBY event in order to put the TV in standby mode and IARM_BUS_PWRMGR_POWERSTATE_STANDBY does the job. IARM_BUS_PWRMGR_POWERSTATE_STANDBY_DEEP_SLEEP & IARM_BUS_PWRMGR_POWERSTATE_STANDBY_LIGHT_SLEEP are other 2 events but i dont see them getting used for hdmicec purposes. 

also for your 2nd query, we don't send the activesource to true; after giving the "HdmiCec_2.1.sendStandbyMessage" command, the TV goes in standby mode (turns off) and needs to wake up again using power button in remote. HdmiCec_2.1.getActiveSourceStatus command verifies the standby mode with the "false" status value.